packages/labs/react/src/create-component.ts

```diff
@@ -114,15 +114,19 @@ const setProperty = <E extends Element>(
   events?: EventNames
 ) => {
   const event = events?.[name];
-  if (event !== undefined) {
+  if (event !== undefined && value !== old) {
     // Dirty check event value.
-    if (value !== old) {
-      addOrUpdateEventListener(node, event, value as (e?: Event) => void);
-    }
-  } else {
-    // But don't dirty check properties; elements are assumed to do this.
-    node[name as keyof E] = value as E[keyof E];
+    addOrUpdateEventListener(node, event, value as (e?: Event) => void);
+    return;
   }
+
+  if (name in HTMLElement.prototype && value === undefined) {
+    node[name as keyof E] = '' as unknown as E[keyof E];
+    return;
```

Comment:
oooh i like that

Reply:
done!

Reply:
> What is this case for, is it trying to match a React special-case? i.e. does React always coerce undefined to empty string for built-in attributes?

This case is for HTMLElement attributes passed as props in React.

They'll fall into the `_elementProps` bucket.

But this edge case is really only a problem on a handful of attributes. Here it's the `id` attribute. If you pass it `undefined` it'll turn into a string `"undefined"`. Now that I'm typing this, i remember a case where an empty string for `id` caused problems.

It might be better to remove the attribute entirely. Will update the PR to reflect that.